### ISSUE TITLE
Reproduction issue: finetuning metrics for "new animal monkey T" do not match the paper`

### ISSUE THREAD

**Issue by GuangShuaiWang** [neuro-galaxy/torch_brain#155]
Hi authors/maintainers,

I’m trying to reproduce the results of POYO-mp on the **“new animal monkey T”** setting, but I’m running into an issue where the **finetuned metrics do not match the metrics reported in the paper**. I would really appreciate some clarification.

### 1. Goal

- Task/setting: `new animal monkey T`
- Goal: finetune the model following the instructions and reproduce the evaluation metrics reported in the paper for this setting in Table 2.

### 2. Experimental setup
- Code:
  - change code form `examples/poyo/train.py`
      - add `_extend_model_vocab` in `DataModule` class.
      - set epoch = 100, batch_size=128 and lr = 1e-4(3.125e-5).

### 3. Metrics I obtain vs. paper

After finetuning and running evaluation, I get the following metrics :

<table>
  <thead>
    <tr>
      <th>Group</th>
      <th>Dataset</th>
      <th>Best R²</th>
      <th>Epoch</th>
      <th>Group Mean &plusmn; Std</th>
    </tr>
  </thead>
  <tbody>
    
    <tr>
     <td rowspan="6" style="text-align: center;">
  <strong>POYO-mp + Finetune</strong><br>
  <strong>0.9012 &plusmn; 0.0271</strong>
      </td>
      <td>t_20130905_center_out_reaching</td>
      <td>0.8635</td>
      <td>58</td>
      <td rowspan="6">0.8337 &plusmn; 0.0267</td>
    </tr>
    <tr>
      <td>t_20130819_center_out_reaching</td>
      <td>0.8288</td>
      <td>85</td>
    </tr>
    <tr>
      <td>t_20130821_center_out_reaching</td>
      <td>0.8520</td>
      <td>82</td>
    </tr>
    <tr>
      <td>t_20130823_center_out_reaching</td>
      <td>0.8417</td>
      <td>89</td>
    </tr>
    <tr>
      <td>t_20130903_center_out_reaching</td>
      <td>0.7792</td>
      <td>83</td>
    </tr>
    <tr>
      <td>t_20130909_center_out_reaching</td>
      <td>0.8367</td>
      <td>74</td>
    </tr>
    
    <tr>
          <td rowspan="6" style="text-align: center;">
  <strong>POYO-mp + Unit ID</strong><br>
  <strong>0.8012 &plusmn; 0.0271</strong>
      </td>
      <td>t_20130905_center_out_reaching</td>
      <td>0.8507</td>
      <td>66</td>
      <td rowspan="6">0.8026 &plusmn; 0.0364</td>
    </tr>
    <tr>
      <td>t_20130819_center_out_reaching</td>
      <td>0.7736</td>
      <td>36</td>
    </tr>
    <tr>
      <td>t_20130821_center_out_reaching</td>
      <td>0.8038</td>
      <td>54</td>
    </tr>
    <tr>
      <td>t_20130823_center_out_reaching</td>
      <td>0.7947</td>
      <td>37</td>
    </tr>
    <tr>
      <td>t_20130903_center_out_reaching</td>
      <td>0.7481</td>
      <td>16</td>
    </tr>
    <tr>
      <td>t_20130909_center_out_reaching</td>
      <td>0.8446</td>
      <td>75</td>
    </tr>
    
    <tr>
           <td rowspan="6" style="text-align: center;">
  <strong>POYO-mp + Finetune</strong><br>
  <strong>0.7759 &plusmn; 0.0471</strong>
      </td>
      <td>t_20130820_random_target_reaching</td>
      <td>0.8270</td>
      <td>90</td>
      <td rowspan="6">0.7036 &plusmn; 0.2390</td>
    </tr>
    <tr>
      <td>t_20130822_random_target_reaching</td>
      <td>0.8309</td>
      <td>88</td>
    </tr>
    <tr>
      <td>t_20130830_random_target_reaching</td>
      <td>0.6907</td>
      <td>92</td>
    </tr>
    <tr>
      <td>t_20130904_random_target_reaching</td>
      <td>0.8472</td>
      <<td>44</td>
    </tr>
    <tr>
      <td>t_20130906_random_target_reaching</td>
      <td>0.1830</td>
      <td>58</td>
    </tr>
    <tr>
      <td>t_20130910_random_target_reaching</td>
      <td>0.8430</td>
      <td>86</td>
    </tr>
    
    <tr>
           <td rowspan="6" style="text-align: center;">
  <strong>POYO-mp + Unit ID</strong><br>
  <strong>0.8105 &plusmn; 0.0561</strong>
      </td>
      <td>t_20130820_random_target_reaching</td>
      <td>0.8084</td>
      <td>51</td>
      <td rowspan="6">0.6831 &plusmn; 0.2321</td>
    </tr>
    <tr>
      <td>t_20130822_random_target_reaching</td>
      <td>0.8005</td>
      <td>80</td>
    </tr>
    <tr>
      <td>t_20130830_random_target_reaching</td>
      <td>0.6516</td>
      <td>11</td>
    </tr>
    <tr>
      <td>t_20130904_random_target_reaching</td>
      <td>0.8271</td>
      <td>45</td>
    </tr>
    <tr>
      <td>t_20130906_random_target_reaching</td>
      <td>0.1822</td>
      <td>24</td>
    </tr>
    <tr>
      <td>t_20130910_random_target_reaching</td>
      <td>0.8289</td>
      <td>32</td>
    </tr>
  </tbody>
</table>

### 4. Main questions

- For the dataset, I am directly using the data processed by the brainset package, so I think the dataset / preprocessing side is unlikely to be the source of the discrepancy.
- I suspect the main issue might be on the hyperparameter side. In particular, I found that the finetuning process is very sensitive to the learning rate. However, using the learning rate of 1e-4 mentioned in the paper, I am not able to match the reported performance.
For finetuning, I adapted the existing train script. I carefully checked the modifications and they seem consistent with the original training setup.
- Could you please give me some hints on where this mismatch is most likely coming from?
For example, is there a specific evaluation script, flag, or configuration that I should use to exactly match the paper’s setup (e.g., a particular --eval_mode or config file for monkey T)?

---

Thanks a lot for your time and for releasing POYO-mp. Any guidance on how to correctly reproduce the “new animal monkey T” finetuning metrics and match them to the paper’s reported metrics would be very helpful.

**Comment by vinamarora8** [comment]
This is likely due to POYO requiring specific a finetuning schedule, which is not implemented in current code. There is a PR to add this but it didn't get merged for a while (https://github.com/neuro-galaxy/torch_brain/pull/81). Maybe try the schedule from this? 

Also, we have moved official POYO implementation has moved to https://github.com/nerdslab/poyo. I have created nerdslab/poyo#1 to track the lack of finetuning scripts . Closing this issue, but feel free to re-open if you want to discuss here.

And apologies for the mess!
